File: papi/lista1/exercicio11.py
nome = input("Por favor digite o se nome: ")

while True:
    try:
        nota1 = float(input("Por favor digite sua nota 1: "))
        break
    except ValueError:
        print("O valor deve ser um número válido, com ponto como separador decimal")

while True:
    try:
        nota2 = float(input("Por favor digite sua nota 2: "))
        break
    except ValueError:
        print("O valor deve ser um número válido, com ponto como separador decimal")

while True:
    try:
        nota3 = float(input("Por favor digite sua nota 3: "))
        break
    except ValueError:
        print("O valor deve ser um número válido, com ponto como separador decimal")

while True:
    try:
        nota4 = float(input("Por favor digite sua nota 4: "))
        break
    except ValueError:
        print("O valor deve ser um número válido, com ponto como separador decimal")

media = (nota1 + nota2 + nota3 + nota4) / 4

print("A média das notas de", nome, "é", media)

if media >= 7:
    print("Aluno aprovado")
else:
    print("Aluno reprovado")

File: papi/lista1/exercicio11.kt
package lista1

fun main() {
    println("Iremos calcular a média de 3 notas")
    var notas: ArrayList<Double> = ArrayList()
    print("Por favor digite o seu nome: ")
    var nome = readLine()

    var cont = 1
    while (cont < 5) {
        while (true) {
            try {
                print("Digite a nota $cont: ")
                val nota = readLine()!!.toDouble()
                if (nota < 0) {
                    throw NumberFormatException()
                }
                notas.add(nota)
                break
            } catch (e: NumberFormatException) {
                println("Por favor digite um número maior que zero válido, usando o ponto como separador decimal")
            }
        }
        cont = cont + 1
    }

    val media = notas.average() 
    println("A média das notas é: $media")

    if (media >= 7) {
        println("Parabéns $nome, você está aprovado!")
    } else {
        println("Você está reprovado $nome")
    }
}